weasyl/searchtag.py

```diff
@@ -17,6 +17,15 @@
 
 _TAG_DELIMITER = re.compile(r"[\s,]+")
 
+_SEARCHTAG_BLACKLIST_REGEXP_PATTERN = re.compile(r"""
+    ^           # Beginning anchor
+    (\w+\*\w+|  # Matches a*a
+    \w{2,}\*|   # Matches aa*
+    \*\w{2,}|   # Matches *aa
+    \w+)        # Matches a, aa, aaa, red_fox, etc.
+    $           # Ending anchor
+    """, flags=re.VERBOSE|re.MULTILINE)
```

Comment:
Why multiline? (If this becomes single-line, also use `\Z` instead of `$` to avoid trailing newlines.)